docs/import.md

```diff
@@ -1,50 +1,89 @@
+You can import a model into Ollama:
 
-A binary GGUF file can be imported directly into Ollama through a Modelfile.
+  * From Safetensors weights; or
+  * From a GGUF file
 
-```dockerfile
-FROM /path/to/file.gguf
-```
+## Importing from Safetensors
+
+Importing your model directly from the Safetensors weights is the easiest way to import a model. Ollama supports importing several different architectures including:
+
+  * Llama (including Llama 2, Llama 3, and Llama 3.1);
+  * Mistral (including Mistral 1, Mistral 2, and Mixtral); and
+  * Gemma (including Gemma 1 and Gemma 2)
+
+This includes importing either the foundation model as well as any fine tuned model which is based upon one of these architectures.
 
-## Import Safetensors
 
-If the model being imported is one of these architectures, it can be imported directly into Ollama through a Modelfile:
+Importing a model requires three steps:
 
- - LlamaForCausalLM
- - MistralForCausalLM
- - GemmaForCausalLM
+  1. Write a Modelfile
+  2. Create your model with `ollama create my-model`
+  3. Test your model with `ollama run my-model`
+
+
+#### Writing a Modelfile
+
+Create a Modelfile using a text editor which includes a `FROM` line which points to the directory with the model weights.
 
 ```dockerfile
 FROM /path/to/safetensors/directory
 ```
 
-For architectures not directly convertable by Ollama, see llama.cpp's [guide](https://github.com/ggerganov/llama.cpp/blob/master/README.md#prepare-and-quantize) on conversion. After conversion, see [Import GGUF](#import-gguf).
+If you create the Modelfile in the same directory as the weights, you can use the line `FROM .`.
 
-## Automatic Quantization
+You can include other settings in your Modelfile such as parameters for the model, the desired chat template, the system prompt, and any license information. Refer to the [Modelfile documentation](https://github.com/ollama/ollama/blob/main/docs/modelfile.md) for more information about Modelfile commands.
 
-> [!NOTE]
-> Automatic quantization requires v0.1.35 or higher.
+#### Create your model with `ollama create`
 
-Ollama is capable of quantizing FP16 or FP32 models to any of the supported quantizations with the `-q/--quantize` flag in `ollama create`.
+After you have created the Modelfile, use the command:
 
-```dockerfile
-FROM /path/to/my/gemma/f16/model
+```bash
+$ ollama create -f Modelfile <model name>
 ```
 
-```shell
-$ ollama create -q Q4_K_M mymodel
-transferring model data
-quantizing F16 model to Q4_K_M
-creating new layer sha256:735e246cc1abfd06e9cdcf95504d6789a6cd1ad7577108a70d9902fef503c1bd
-creating new layer sha256:0853f0ad24e5865173bbf9ffcc7b0f5d56b66fd690ab1009867e45e7d2c4db0f
-writing manifest
-success
+If the model architecture of your weights is supported, Ollama will take a few moments to import the weights. It will also attempt to find the correct _chat template_ inside the model's configuration data. You can override the template using the `TEMPLATE` command in your Modelfile.
```

Comment:
```suggestion
If the model architecture of your weights is supported, Ollama will take a few moments to import the weights. It will also attempt to find the correct _chat template_ inside the model's `tokenizer_config.json` file. You can override the template using the `TEMPLATE` command in your Modelfile.
```

Reply:
I purposely removed this from the original document because I don't think it's useful. We don't tell you how to modify the `tokenizer_config.json` file to get it to pick up the template correctly, nor would we want to (since you can just using the `TEMPLATE` command to do that). Ultimately the mechanism isn't important here.

Reply:
I agree - makes sense!